doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,356 @@
+
+To evaluate the custom rule against the training, you would do:
+
+.. code:: python
+
+    from sagemaker.debugger import Rule
+
+    custom_gradient_rule = Rule.custom(
+        name='MyCustomRule',
+        image_uri='864354269164.dkr.ecr.us-east-1.amazonaws.com/sagemaker-debugger-rule-evaluator:latest',
+        instance_type='ml.t3.medium', # instance type to run the rule evaluation on
+        source='rules/custom_gradient_rule.py', # path to the rule source file
+        rule_to_invoke='CustomGradientRule', # name of the class to invoke in the rule source file
+        volume_size_in_gb=30, # EBS volume size required to be attached to the rule evaluation instance
+        collections_to_save=[CollectionConfig("gradients")], # collections to be analyzed by the rule
+        rule_parameters={
+          "threshold": "20.0" # this will be used to initialize 'threshold' param in your rule constructor
+        }
+    )
+
+    estimator = TensorFlow(
+        role=role
+        train_instance_count=1,
+        train_instance_type=train_instance_type,
+        rules=[
+            custom_gradient_rule
+        ]
+    )
+
+While initializing the custom rule through ``Rules.custom()``, for the value of ``source`` you can optionally specify a valid S3 location as well.
+
+
+Capture real-time TensorBoard data from the debugging hook
+==========================================================
+
+In addition to emitting and storing the debugging data useful for analyses, the debugging hook is also capable of emitting `TensorBoard <https://www.tensorflow.org/tensorboard>`__ data for you to point your TensorBoard application at and visualize.
+
+To enable the debugging hook to emit TensorBoard data, you'd need to specify the new option ``TensorBoardOutputConfig`` as:
+
+.. code:: python
+
+    from sagemaker.debugger import TensorBoardOutputConfig
+
+    tensorboard_output_config = TensorBoardOutputConfig(
+        s3_output_path='s3://path/for/tensorboard/data/emission',
+        container_local_output_path='/local/path/for/tensorboard/data/emission'
+    )
+
+    estimator = TensorFlow(
+        role=role
+        train_instance_count=1,
+        train_instance_type=train_instance_type,
+        tensorboard_output_config=tensorboard_output_config
+    )
+
+When you call ``fit()`` on the ``estimator``, the debugging hook will upload the generated TensorBoard data in near real-time to an S3 path derived from the value of ``s3_output_path`` provided in the configuration:
+
+.. code::
+
+    s3://{destination-bucket-prefix}/tensorboard-output/
+
+To access the S3 path where the tensorboard data is stored, you can do:
+
+.. code:: python
+
+    tensorboard_s3_output_path = estimator.latest_job_tensorboard_artifacts_path()
+
```

Comment:
Note: Tensorboard output can slow down your training job.

Reply:
you'd need to specify -> you need to specify

Reply:
* [@Vikas-kum] Added.
* [@aws-rbs] Fixed.